pkg/stacker/publisher.go

```diff
@@ -261,7 +234,92 @@ func (p *Publisher) publishArtifact(path, mtype, registry, repo, subject string,
 		return errors.Errorf("unable to upload manifest, url:%s", regUrl)
 	}
 
-	log.Infof("artifact '%s' sucessfully uploaded to url:%s", path, regUrl)
+	log.Infof("Copying artifact '%s' done", path)
+
+	return nil
+}
+
+func distspecURL(registry, repo, tag string, skipTLS bool) string {
+	var url string
+
+	if skipTLS {
+		url = fmt.Sprintf("http://%s/v2%s/manifests/%s", registry, strings.Split(repo, ":")[0], tag)
+	} else {
+		url = fmt.Sprintf("https://%s/v2%s/manifests/%s", registry, strings.Split(repo, ":")[0], tag)
+	}
+
+	return url
+}
+
+func uploadBlob(registry, repo, path, username, password string, reader io.Reader, size int64, dgst *godigest.Digest, skipTLS bool) error {
+	// upload with POST, PUT sequence
+	var regUrl string
```

Comment:
Not asking you to do that here for this pr, but just a comment that tracking the regUrl and subject as instances of a new custom url class, which knows whether it's skipTLS or not and tracks repo etc, and has .BlobURL and .ManiestURL both returning a string url, would be neat here.

Reply:
Would be neat I suppose.